web/src/components/repo/build/BuildLog.vue

```diff
@@ -205,87 +264,34 @@ export default defineComponent({
           repo.value.name,
           build.value.number,
           proc.value.ppid,
-          (l) => {
-            loadedLogs.value = true;
-            term.value.write(l.out, () => {
-              if (autoScroll.value) {
-                term.value.scrollToBottom();
-              }
-            });
+          (line) => {
+            if (line?.proc !== proc.value?.name) {
+              return;
+            }
+            writeLog({ index: line.pos, text: line.out, time: line.time });
+            flushLogs(true);
```

Comment:
Instead of passing a `scroll` argument to `flushLogs` we could directly add the scroll call here and remove it from `flushLogs`.

```suggestion
            flushLogs();
            if (autoScroll.value) {
              scrollDown();
            }
```

Reply:
not really, as `flushLog` will be debounced to my understanding it could be called after specified interval only, so scrolling can't really be moved out otherwise it could be that lines will be added for display only after scroll has already been adjusted.

Reply:
Good point, missed that.